code/modules/urist/modules/ghost_powers.dm

```diff
@@ -20,7 +36,79 @@
 /mob/living/carbon/attack_ghost(mob/observer/ghost/user as mob)
 	. = ..()
 	if(src.sleeping && src.client && round_is_spooky())
-		var/dream = sanitize(input(user, "Enter someone's dreams", "Dream") as text|null)
+		user.ghost_dream_invasion(src, TRUE)
+
+
+/mob/observer/ghost/proc/ghost_dream_invasion(var/mob/living/carbon/human/target = null, var/target_optional = TRUE)
+	// Write messages into people's dreams. Spooooookyyyyy.
+	// To make it more Fun, you have to use normal dream strings, BUT:
+	// a) There's a small chance you get to write anything you want instead
+	// b)
+
+	set category = "Cult"
+	set name = "Invade dreams"
+	set desc = "Insert a message into someone's dream. Remember, no IC in OOC or OOC in IC."
+
+	if(!round_is_spooky())
+		to_chat(src, SPAN_NOTICE("The veil is not thin enough to let you use this power."))
+		return
+
+	if(!ghost_ability_check())
+		return
+
+	var/mob/living/carbon/human/choice = target
+
+	if(!istype(choice) && target_optional)
+		// if target_optional is FALSE, we MUST use the target
+		// otherwise, allow picks from adjacent sleepers
+		var/list/mob/living/carbon/human/choices = list()
```

Comment:
Okay, legit question. Does this actually work? Don't think I've ever seen anyone put a type path in a list before like this

Reply:
It does. In fact, this is a direct copypaste of code already in the repo: https://github.com/UristMcStation/UristMcStation/blob/4357cf02972e85efbe6bad3a429e49c34fafcea8/code/game/gamemodes/cult/ghosts.dm#L209

IIRC, this is an undocumented syntax; it doesn't CURRENTLY do anything different than a pure list(), but is reserved for the future as a potential optimization.

Reply:
Interesting. I'll have to keep my eye on that :eyes: